### PR TITLE
Set a check on length of password field in sign up form.

### LINKED ISSUE #578: No check on length of password in sign-up form
## Description
As a user I must keep my password in some text limit so that the security can be improved.

## Mocks

screenshot from 2018-02-02 14-47-55

### Update [Required]
- [ ] Implement min_length and max_length check on password

## Definition of Done
- [ ] All of the required items are completed.
- [ ] Approval by 1 mentor.

## Estimation
5 minutes

### PR DESCRIPTION
**Pull request by sammy1997**
# Description
Set a max and min length for password input field in registration / sign up form 

Fixes #578 

password

# Type of Change:
- Code
- Bug fix (non-breaking change which fixes an issue)

# Checklist:

- [ ] My PR follows the style guidelines of this project
- [ ] I have performed a self-review of my own code or materials
- [ ] My changes generate no new warnings 
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by sammy1997** [comment]
@tapasweni-pathak Please review and merge this.

**Comment by sammy1997** [comment]
@tonythomas01 Please review this.

**Comment by tonythomas01** [comment]
Looks good, even though - from next time I would better see a unit test associated with these kinds of changes.

**Comment by sammy1997** [comment]
Sure :+1: @tonythomas01 .

**Comment by sammy1997** [comment]
@tonythomas01 If it's okay, can you please merge this PR? Thanks.

**Comment by sammy1997** [comment]
@tonythomas01 Please merge this.

File: vms/registration/forms.py [hunk 1]
```diff
@@ -5,7 +5,7 @@
 
 class UserForm(forms.ModelForm):
     # password not visible when user types it out
-    password = forms.CharField(widget=forms.PasswordInput())
+    password = forms.CharField(widget=forms.PasswordInput(), min_length=8, max_length=25)
```
Comment (tapasweni-pathak) [inline vms/registration/forms.py]: could you also add a message for the user that they won't be able to have a password < 8 length?

Comment (tapasweni-pathak) [inline vms/registration/forms.py]: could you also add a message for the user that they won't be able to have a password < 8 length?

**Comment by sammy1997** [comment]
@tapasweni-pathak Please review this. I have added a message in the form

**Comment by sammy1997** [comment]
@tapasweni-pathak @tonythomas01 Please review this

File: vms/registration/templates/registration/signup_volunteer.html [hunk 2]
```diff
@@ -42,7 +42,7 @@ <h2>{% trans "Sign Up" %}</h2>
                     {% trans "Password" %}<span class="asteriskField">*</span>
                 </label>
                 <div class="controls">
-                    <input id="id_password" class="textinput textInput form-control" type="password" placeholder="{% blocktrans %}Password{% endblocktrans %}" name="usr-password">
+                    <input id="id_password" class="textinput textInput form-control" type="password" placeholder="{% blocktrans %}Password ( password length should be at least 8 characters long){% endblocktrans %}" name="usr-password">
```
Comment (tapasweni-pathak) [inline vms/registration/templates/registration/signup_volunteer.html]: Could you add a error message for >max_length and not placeholder text? For <min_length VMS throws error message.

Comment (tapasweni-pathak) [inline vms/registration/templates/registration/signup_volunteer.html]: Could you add an error message and not placeholder text?

**Comment by sammy1997** [comment]
@tapasweni-pathak I have made the requested changes. Please review this.

**Comment by sammy1997** [comment]
@Buddhiprabha @tapasweni-pathak Please review this

**Comment by sammy1997** [comment]
@kamsuri Please review this PR.

**Comment by prachi1210** [comment]
can you write unit tests alongside this?

**Comment by sammy1997** [comment]
Ok @prachi1210 . I will write the tests and ping you

File: vms/registration/forms.py [hunk 3]
```diff
@@ -5,8 +5,16 @@
...
```
Comment (ayushgarg1804) [inline vms/registration/forms.py]: Any reason for 25 max length? Usually its best practice for maximum length of password field to be of 128 characters.

File: vms/registration/templates/registration/signup_volunteer.html (hunk 2 above)
Comment (ayushgarg1804) [inline vms/registration/templates/registration/signup_volunteer.html]: Add maximum length error message too for passwords and as @tapasweni-pathak said an error message accordingly and not placeholder text.

**Comment by ayushgarg1804** [comment]
@sammy1997 Are you still working on this or can it be closed for another member to solve?

**Comment by sammy1997** [comment]
@ayushgarg1804 I was working on Sysbot, so sorry for the delay. Maybe a weeks time will be good. If I don't get time till then I will release the issue and close the PR

**Comment by ayushgarg1804** [comment]
@sammy1997 Okay, assigning you to the issue for now. Do inform if you won't be able to do it.

**Comment by anjali-dhanuka** [comment]
@sammy1997 Do u plan to work on it?

**Comment by sammy1997** [comment]
@anjali-dhanuka I am not getting time for this due to Sysbot. So better to assign someone else. Sorry for the delay

**Pull request closed without merging**